test/TokenUriResolver.t.sol

```diff
@@ -0,0 +1,287 @@
+// SPDX-License-Identifier: UNLICENSED
```

Comment:
I'd add tests covering getUri branches ( resolver == address(0), resolver != 0 but reverting, resolver != 0  and not reverting), same in setTokenUriResolverForProject (resolver == 0  or not)